internal/model/rwfolder.go

```diff
@@ -874,16 +876,26 @@ func (p *rwFolder) shortcutFile(file protocol.FileInfo) (err error) {
 	t := time.Unix(file.Modified, 0)
 	err = os.Chtimes(realName, t, t)
 	if err != nil {
-		if p.lenientMtimes {
+		// First try using virtual mtimes
+		info, errStat := os.Stat(realName)
+
+		if errStat == nil {
 			err = nil
-			// We accept the failure with a warning here and allow the sync to
-			// continue. We'll sync the new mtime back to the other devices later.
-			// If they have the same problem & setting, we might never get in
-			// sync.
-			l.Infof("Puller (folder %q, file %q): shortcut: %v (continuing anyway as requested)", p.folder, file.Name, err)
+			p.virtualMtimeRepo.UpdateMtime(file.Name, info.ModTime(), t)
 		} else {
-			l.Infof("Puller (folder %q, file %q): shortcut: %v", p.folder, file.Name, err)
-			return
+			l.Infof("Puller (folder %q, file %q): shortcut: unable to stat file: %v", p.folder, file.Name, errStat)
+
+			if p.lenientMtimes {
```

Comment:
Yes, I had that thought as well but wasn't sure if the config reader would complain about an unrecognized option or if it would just ignore it.  (I didn't want to break backwards-compatibility by removing it.)

Currently this code will only get hit if the stat on the new file fails, which could happen if some external process unlinks/moves it between when we create it and when we stat it.  We should handle this case somehow, but I'm sure it doesn't require the "lenientMtimes" option.